packages/components/src/checkbox/tests/jest/Checkbox.test.tsx

```diff
@@ -0,0 +1,108 @@
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveClass("hop-Checkbox");
+        expect(element).toHaveClass("test");
+    });
+
+    it("should support custom style", () => {
+        render(<Checkbox marginTop="stack-sm" style={{ marginBottom: "13px" }} >option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveStyle({ marginTop: "var(--hop-space-stack-sm)", marginBottom: "13px" });
+    });
+
+    it("should support DOM props", () => {
+        render(<Checkbox data-foo="bar">option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox").closest("label");
+        expect(element).toHaveAttribute("data-foo", "bar");
+    });
+
+    it("should support slots", () => {
+        render(
+            <CheckboxContext.Provider value={{ slots: { test: { "aria-label": "test" } } }}>
+                <Checkbox slot="test">option 1</Checkbox>
+            </CheckboxContext.Provider>
+        );
+
+        const checkbox = screen.getByRole("checkbox");
+        const element = checkbox.closest("label");
+        
+        expect(element).toHaveAttribute("slot", "test");
+        expect(checkbox).toHaveAttribute("aria-label", "test");
+    });
+
+    it("should support refs", () => {
+        const ref = createRef<HTMLLabelElement>();
+        render(<Checkbox ref={ref}>option 1</Checkbox>);
+
+        expect(ref.current).not.toBeNull();
+        expect(ref.current instanceof HTMLLabelElement).toBeTruthy();
+    });
+
+    it("should be focused on render when autofocus is true", async () => {
+        render(<Checkbox autoFocus >option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox");
+        await waitFor(() => expect(element).toHaveFocus());
+    });
+
+    it("should not be focused on render when autofocus is true and the checkbox is disabled", async () => {
+        render(<Checkbox isDisabled autoFocus >option 1</Checkbox>);
+
+        const element = screen.getByRole("checkbox");
+        await waitFor(() => expect(element).not.toHaveFocus());
+    });
+
+    it("should prevent onChange when the checkbox is disabled", async () => {
+        const handler = jest.fn();
+        const user = userEvent.setup();
+
+        render(
+            <Checkbox
+                isDisabled
+                onChange={handler}
+            >
+            Disabled button
+            </Checkbox>
+        );
+
+        const element = screen.getByRole("checkbox");
+        await user.click(element);
+
+        expect(handler).not.toHaveBeenCalled();
+    });
+
+    // ***** Api *****
+
+    it("should be focused on render when the focus api is called", async () => {
+        const ref = createRef<HTMLLabelElement>();
+
+        render(<Checkbox ref={ref}>option 1</Checkbox>);
+        const checkboxElement = ref.current?.querySelector("input[type='checkbox']") as HTMLInputElement;
```

Comment:
this code feel odd. 

I think we should wait for the next release, and change the ref to inputref. So focusing the inputRef would work instead of using the querySelector

https://github.com/adobe/react-spectrum/pull/5967